geoapps/utils/surveys.py

```diff
@@ -377,3 +377,37 @@ def split_dcip_survey(
             line_surveys.append(line_survey)
 
     return line_surveys
+
+
+def get_unique_locations(survey: BaseSurvey) -> np.ndarray:
+    if survey.source_list:
+        locations = []
```

Comment:
This seems like it could be handled by all geoh5py classes with a source_list attr.  This way we can simplify/remove the logic

Reply:
..otherwise pull out the guts of the if in another utility function and call here (get_source_receiver_locations?)

Reply:
It's a SimPEG class, not geoh5py.

Reply:
right, didn't realize..